device_backends/src/NumericAddressedBackend.cc

```diff
@@ -57,6 +57,30 @@ namespace ChimeraTK {
     }
   }
 
+  /* Throw exception if called directly and not implemented by backend */
+  void NumericAddressedBackend::read([[maybe_unused]] uint8_t bar,
```

Comment:
@mhier suggested this default 32-bit implementation, so that newer backends (like the xdma one) don't have to implement both 32- and 64-bit interfaces. I can also imagine the 32-bit default implementation just calling the 64-bit interface, but if that accidentally is also not implemented, it would call the 32-bit one again, resulting in a infinite loop.

Reply:
What we need is a default implementation of the new interface (64 bit address), which calls the old interface (32 bit address). The old interface also needs a default implementation, to avoid that backends still need to implement the old interface. To avoid the lack of any implementation to stay somewhat unnoticed, the old interface's default implementation should terminate in an error. There is no infinite loop in this construction.